pkg/model/application.go

```diff
@@ -89,8 +89,8 @@ func (a *Application) SetUpdatedAt(t int64) {
 	a.UpdatedAt = t
 }
 
-func (k *ApplicationKind) CompatiblePlatformProviderType() PlatformProviderType {
-	switch *k {
+func (x *ApplicationKind) CompatiblePlatformProviderType() PlatformProviderType {
```

Comment:
Why do we need to update this?

Reply:
To resolve this error.

> pkg/model/application.go:92:27: ST1016: methods on the same type should have the same receiver name (seen 1x "k", 1x "x") (stylecheck)

By the way, I reconsidered the naming "k" is more appropriate naming "x", so I unify the naming with A and use a value receiver for the enum by following other implementations.

https://github.com/pipe-cd/pipecd/pull/4638/commits/e8103700923e410bacbceff6f836b64c87469038

Reply:
Got it, thanks 👍

Reply:
I see. Fixed it👍


https://github.com/pipe-cd/pipecd/pull/4638/commits/5bb575b50aad98e9d0a34d5b43654a1bb8a904ed